Feat/generated reference docs (#1294)
tinacms/tinacms.org: content/docs/tina-folder/overview.md

---
title: The .tina folder
id: /docs/tina-folder/overview/
# next: '/docs/features/data-fetching'
---

## Overview

The `.tina` folder is where all of the TinaCMS configuration and customization is. This is the place where the schema is defined and where the editing experience can be customized.

The `.tina` folder normally is at the root of the repo (unless a [mono repo setup](/docs/tina-cloud/faq/#does-tina-cloud-work-with-monorepos) is being used) and has the following folder structure:

> - [`schema.{ts,tsx,js}`](/docs/schema/)

Author: Yup totally

- `components` folder
  - `TinaProvider.js`
  - `TinaDynamicProvider.js`
- `queries` folder (*optional*)
- `__generated___` folder
  - `_graphql.json`
  - `_lookup.json`
  - `_schema.json`
  - `frags.gql`
  - `queries.gql`
  - `schema.gql`

## `schema.{ts,tsx,js}`

> This is where the schema is defined and must be the default export of this file. This is also where the tina config for the site can be defined.

Reviewer: ```suggestion
This file contains the definition for the project's schema, as well as the Tina configuration object. The schema must be the default export of this file.
See our ["Tina Cloud Starter"](https://github.com/tinacms/tina-cloud-starter/blob/main/.tina/schema.ts) for an example of how this file is used
```

```
import { defineSchema, definedConfig } from 'tinacms'

const schema = defineSchema({...})
export const config = definedConfig({ schema: schema, ...})
export default schema
```

## `components` folder

The folder is generated by the tina Init process and contains two files. If this is not already in your site an example can be [seen here](https://github.com/tinacms/tina-cloud-starter/tree/main/.tina/components). These two files (more details below) are responsible for the importing the config from the schema file, passing the config to the tina provider, and dynamically loading tina so it is only loaded in edit mode. 

### `TinaProvider.js`

This file handles importing the config and TinaProvider.

### `TinaDynamicProvider.js`

This file lazy loads the `TinaProvider.js` to ensure that tina is only loaded in edit mode and does not effect your production bundle.

## `queries` folder (*optional*)

The `queries` folder is optional and is used by the [experimental generated client](/docs/graphql/client/). The queries that are defined in this folder will automatically be attached to the generated client when it is built.

## `__generated___` folder

> This is where all the files that are generated durning the schema build process are put. Generally speaking, these files **must be checked into source control and pushed to github** with a couple of exceptions. The reason for this is that Tina cloud looks at the generated files in order to resolve documents.

Author: These files to not need to be checked into source control
- schema.gql
- queries.gql
- frags.gql

### `_graphql.json`

The Graphql AST (represented in JSON). This is used by Tina Cloud so it must be pushed to github

### `_lookup.json`

This is a lookup file that is used to resolve document names. (This must also be pushed to github)

### `_schema.json`

The Graphql Schema AST (represented in JSON). This is used by Tina Cloud so it must be pushed to github

### `frags.gql`

> This the the raw graphql fragments that are generated and used by the [experimental generated client](/docs/graphql/client/). This does not need to be checked into source control. 

Author: This file is used in the build process of the client. Since the client is built locally it is only needed locally and not in production.  The actual client lives in `.tina/__generated__/types.ts` and that will need to be pushed to github. Ill update the docs to make it more clear.

### `queries.gql`

> This the the raw graphql queries that are generated and used by the [experimental generated client](/docs/graphql/client/). This does not need to be checked into source control. 

Reviewer 2: ```suggestion
This file contains the raw graphql queries that are generated and used by the [experimental generated client](/docs/graphql/client/). Does not need to be checked into source control. 
```

> This the the raw graphql queries that are generated and used by the [experimental generated client](/docs/graphql/client/). This does not need to be checked into source control. 

Reviewer: ```suggestion
This file contains the raw graphql queries that are generated and used by the [experimental generated client](/docs/graphql/client/). This does not need to be checked into source control. 
```

### `schema.gql` 

> This the the raw graphql schema. This does not need to be checked into source control. 

Reviewer: ```suggestion
This file contains the raw GraphQL schema. This does not need to be checked into source control. 
```
(There's a few other spots where graphql should be GraphQL)

> This the the raw graphql schema. This does not need to be checked into source control. 

Reviewer 2: ```suggestion
This file contains the raw graphql schema. Does not need to be checked into source control.
```
